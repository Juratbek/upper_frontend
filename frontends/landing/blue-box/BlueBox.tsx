import { UpRight } from 'components/icons/UpRight';
import { Clickable } from 'components/lib';
import { useAuth } from 'hooks';
import { FC } from 'react';

import { LogosBox } from '../logos-box/LogosBox';
import classes from './BlueBox.module.scss';

export const BlueBox: FC = () => {
  const { openLoginPage } = useAuth();

  return (
    <div className={classes.container}>
      <h1 className={classes.heading}>upper bilan yanada yuqoriroq</h1>
      <div className={classes.body}>
        <Clickable className={classes.start} onClick={() => openLoginPage()}>
          <span className={classes['start-text']}>Boshladik</span>
          <span className={classes['up-right-icon']}>
            <UpRight />
          </span>
        </Clickable>
        <LogosBox />
      </div>
    </div>
  );
};
