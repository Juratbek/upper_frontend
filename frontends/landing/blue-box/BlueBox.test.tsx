import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ReduxWrapper } from 'tests';
import { vi } from 'vitest';

import { BlueBox } from './BlueBox';

const mocks = vi.hoisted(() => ({
  openLoginPage: vi.fn(),
}));

vi.mock('hooks', async (importOriginal) => {
  const original = await importOriginal<typeof import('hooks')>();

  return {
    ...original,
    useAuth: () => ({ openLoginPage: mocks.openLoginPage }),
  };
});

const Component = (
  <ReduxWrapper>
    <BlueBox />
  </ReduxWrapper>
);

describe('BlueBox', () => {
  it('snapshot', () => {
    render(Component);
    expect(document.body).toMatchSnapshot();
  });

  it('opens login page on login button click', async () => {
    render(Component);

    const loginButton = screen.getByRole('button', { name: 'Boshladik' });
    userEvent.click(loginButton);

    await waitFor(() => {
      expect(mocks.openLoginPage).toHaveBeenCalledTimes(1);
    });
  });
});
