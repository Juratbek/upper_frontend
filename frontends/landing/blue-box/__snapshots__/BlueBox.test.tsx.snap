// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlueBox > snapshot 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <h1
        class="heading"
      >
        upper bilan yanada yuqoriroq
      </h1>
      <div
        class="body"
      >
        <button
          class="root start"
        >
          <span
            class="start-text"
          >
            Boshladik
          </span>
          <span
            class="up-right-icon"
          >
            <svg
              fill="none"
              height="20"
              role="icon"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M1.5 2H17.5M17.5 2V18M17.5 2L1.5 18"
                stroke="#000C1A"
                stroke-width="4"
              />
            </svg>
          </span>
        </button>
        <div
          class="container"
        >
          <img
            alt="vs code"
            class="image"
            data-nimg="1"
            decoding="async"
            height="120"
            loading="lazy"
            src="/_next/image?url=%2Flogos%2Fvs_code.png&w=256&q=75"
            srcset="/_next/image?url=%2Flogos%2Fvs_code.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fvs_code.png&w=256&q=75 2x"
            style="color: transparent; --i: 0;"
            width="120"
          />
          <img
            alt="vs code"
            class="image"
            data-nimg="1"
            decoding="async"
            height="120"
            loading="lazy"
            src="/_next/image?url=%2Flogos%2Fchat-gpt.png&w=256&q=75"
            srcset="/_next/image?url=%2Flogos%2Fchat-gpt.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fchat-gpt.png&w=256&q=75 2x"
            style="color: transparent; --i: 1;"
            width="120"
          />
          <img
            alt="vs code"
            class="image"
            data-nimg="1"
            decoding="async"
            height="120"
            loading="lazy"
            src="/_next/image?url=%2Flogos%2Fjs.png&w=256&q=75"
            srcset="/_next/image?url=%2Flogos%2Fjs.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fjs.png&w=256&q=75 2x"
            style="color: transparent; --i: 2;"
            width="120"
          />
          <img
            alt="vs code"
            class="image"
            data-nimg="1"
            decoding="async"
            height="120"
            loading="lazy"
            src="/_next/image?url=%2Flogos%2Fadobe.png&w=256&q=75"
            srcset="/_next/image?url=%2Flogos%2Fadobe.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fadobe.png&w=256&q=75 2x"
            style="color: transparent; --i: 3;"
            width="120"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
